# Assignment 3
### Due Tuesday 9/23. Do four of five.

In [1]:
import pandas as pd
import numpy as np

1. 
- Open the NHANES (or Ames prices or college completion datasets, if you prefer)
- Find two categorical variables of interest (there are 198, and short descriptions are given in the `nhanes_meta_17_18.csv` file). Investigate their missing values (you don't have to focus on missing values for this analysis like we did with police use of force, but always be aware of how dirty the data are)
- Compute a contingency table for your categorical $X$ and $Y$
- Discuss any interesting patterns (or lack of one) that you observe

In [10]:
data = pd.read_csv(r"C:\Users\crush\OneDrive\Desktop\DS 5030\understanding_uncertainty\data\nhanes_data_17_18.csv")

C:\Users\crush\AppData\Local\Temp\ipykernel_31704\3076742054.py:1: DtypeWarning: Columns (142) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv(r"C:\Users\crush\OneDrive\Desktop\DS 5030\understanding_uncertainty\data\nhanes_data_17_18.csv")


In [ ]:
# investigate missing values
print(f"Missing Values for Diet: {data['HowHealthyIsTheDiet'].isna().sum()}")
print(f"Missing Values for Milk: {data['Past30DayMilkProductConsumption'].isna().sum()}")

Missing Values for Diet: 2540
Missing Values for Milk: 1


In [ ]:
# contingency table from notes
pd.crosstab(data['Past30DayMilkProductConsumption'], data['HowHealthyIsTheDiet'],normalize='index')

HowHealthyIsTheDiet                                 Excellent      Fair  \
Past30DayMilkProductConsumption                                           
Never                                                0.086739  0.256047   
Often-once a day or more?                            0.093381  0.243499   
Rarely-less than once a week                         0.054795  0.290894   
Sometimes-once a week or more but less than onc...   0.064555  0.277944   
Varied                                               0.047619  0.333333   

HowHealthyIsTheDiet                                     Good      Poor  \
Past30DayMilkProductConsumption                                          
Never                                               0.376147  0.078399   
Often-once a day or more?                           0.389480  0.060284   
Rarely-less than once a week                        0.394037  0.078163   
Sometimes-once a week or more but less than onc...  0.403467  0.059773   
Varied                                              0.428571  0.095238   

HowHealthyIsTheDiet                                 Very good  
Past30DayMilkProductConsumption                                
Never                                                0.202669  
Often-once a day or more?                            0.213357  
Rarely-less than once a week                         0.182111  
Sometimes-once a week or more but less than onc...   0.194262  
Varied                                               0.095238

**Patterns:** According to this contingency table, a "Good" diet seems to be the most common observation no matter the milk consumption over the past 30 days. Moreover, a "Fair" diet is in second place no matter the milk consumption habits of the respondent. One thing that deviates from a pattern is the percentage of observations that have both "Varied" milk consumption and a "Very Good" diet. Generally, no matter the milk consumption, the fraction of onbservations also having a "Very Good" diet is approxiamtely 1 in 5. However, for "Varied" milk consumption, the proportion of "Very Good" diets is half of that, at 10%. There also seems to be a trend that for respondents with "Varied" milk consumption, they tend to rate their diet more negatively with the "Varied" category being the highest in "Poor" at 9.5% and lowest in "Excellent" and "Very Good" at 4.7% and 9.5%, respectively. Most importantly, there doesn't seem to be any difference in the respondent's diet when their milk consumption is "Often" or "Never."

2. 
- Open the NHANES dataset
- Find a categorical and numeric variable of interest (there are 198, and short descriptions are given in the `nhanes_meta_17_18.csv` file). Investigate their missing values (you don't have to focus on missing values for this analysis, but always be aware of them)
- Make descriptive tables and grouped kernel density plots to represent the variation in your numeric $Y$ conditional on your categorical $X$
- Discuss any interesting patterns (or lack of one) that you observe

3. 
We showed that the mean and median could be discovered by minimizing various kinds of loss functions; this is what machine learning is. To make a prediction $\hat{y}(z)$ of $Y$ when $X=z$, minimize the mean squared error:
$$
MSE(\hat{y}(z)) = \dfrac{1}{N} \sum_{i=1}^N \left\lbrace y_i - \hat{y}(z) \right\rbrace^2 \frac{1}{h}k\left(\frac{z-x_i}{h}\right)
$$
Show that the solution to this problem is the LCLS/Naradaya-Watson estimator.

4. 
- Write a class or set of functions that implement the LCLS/Naradaya-Watson estimator, using the Silverman plug-in estimate for the conditioning variable $X$ as the bandwidth.
- From one of the course data sets, find two numeric variables of interest, analyze their relationship with the the LCLS/Naradaya-Watson estimator, and discuss your results.

5. 
- In any of the available data sets, investigate the relationships between pairs of variables $(X,Y)$ with a scatterplot and CEF (for example, price on area)
- Is this relationship plausibly causal, or are there missing variables that might explain at least part of the relationship between your variables? These can be "conceptual" rather than "practical"; for example, 'talent' or 'grit' probably explain education outcomes, but are almost impossible to measure. We are asking whether there are hypothetical **threats to causal identification** of the effect of $X$ on $Y$.
- Explain how, regardless of the threat to causal identification, you can still use your model to predict $Y$ given $X$, as long as you don't intervene in the system to control the outcome